### DISCUSSION TITLE
How to change the transparence of specific grains

### DISCUSSION THREAD

**Discussion by kun-Jiang** [mtex-toolbox/mtex#2037] [Ask Anything] [answered]
Hi, all
I have selected certain grains and outlined them in red. However, it is not very easy to identify these grains. Therefore, I would like to reduce the transparency of other grains. Could you guide me on which property to adjust for this purpose?

![Trangranular](https://github.com/mtex-toolbox/mtex/assets/53817143/3fac8f3e-0835-43a3-8fcf-8db9912112a9)

**Comment by zmichels** [comment]
Try adding an argument pair to your plotting command with:

'FalceAlpha', 0.5

Try other values <1.

This should reduce the plotting opacity.

**Reply by kun-Jiang** [reply]
This command seems to have no effect, although it does not cause any runtime errors. The following is my code

> plotx2west % this command tells MTEX to plot the x coordinates increasing to the east (left)
figure
plot(ebsd('Ni (Nickel)'),ebsd('Ni (Nickel)').orientations,'micronbar','off','FalceAlpha', 0.1)
% Remove the grain reconstruction in crack region
[grains,ebsd.grainId] = calcGrains(ebsd,'unitCell','angle',10*degree,'theshold',10*degree);
grains = grains(grains.grainSize > 10);
hold on
% plot(grains.boundary,'lineWidth',0.5);
% hold off
% Transgranular fracture
TGid = [211 39 307 237 266 198 238 185 209 181 207 223 163 108];
TranGrain = grains(TGid);
plot(TranGrain.boundary,'lineWidth',2,'linecolor','red');

**Reply by kun-Jiang** [reply]
Hi zmichels
Thank you for your advice. I carefully reviewed the documentation on the mtex official website, and there was a spelling error in the command you provided. Nevertheless, it was generally correct, and I can now successfully meet my requirements. Following is the correct code

> plotx2west % this command tells MTEX to plot the x coordinates increasing to the east (left)
figure
plot(ebsd('Ni (Nickel)'),ebsd('Ni (Nickel)').orientations,'micronbar','off','FaceAlpha', 0.2)
% Remove the grain reconstruction in crack region
[grains,ebsd.grainId] = calcGrains(ebsd,'unitCell','angle',10*degree,'theshold',10*degree);
grains = grains(grains.grainSize > 10);
hold on
% plot(grains.boundary,'lineWidth',0.5);
% hold off
% Transgranular fracture
TGid = [211 39 307 237 266 198 238 185 209 181 207 223 163 108];
TranGrain = grains(TGid);
plot(TranGrain.boundary,'lineWidth',2,'linecolor','red');
% Highlight the grains
plot(ebsd(TranGrain),ebsd(TranGrain).orientations,'micronbar','off','FaceAlpha', 1)

![image](https://github.com/mtex-toolbox/mtex/assets/53817143/4ae30ed2-06f4-4eb4-b696-ad9eb0f72d41)

**Comment by kilir** [comment, marked as answer]
Hi
tranparency against white background does hardly make any difference, also plotting the transparent thing first has no effect on what comes above.

Some examples to highlight grains 40-60:
```
mtexdata titanium

grains =ebsd.calcGrains;

plot(ebsd,ebsd.prop.ci)
mtexColorMap white2black
hold on
plot(grains([1:39 61:end]),grains([1:39 61:end]).meanOrientation,'FaceAlpha',0.5)
hold on
plot(grains(40:60),grains(40:60).meanOrientation)
hold off

%%
plot(ebsd,'faceColor','k')
mtexColorMap white2black
hold on
plot(grains([1:39 61:end]),grains([1:39 61:end]).meanOrientation,'FaceAlpha',0.1)
hold on
plot(grains(40:60),grains(40:60).meanOrientation)
hold off
```

Cheers,
Rüdiger

**Reply by kun-Jiang** [reply]
Hi Rüdiger
Thank you for your response. The code you provided has been very helpful, and the issue has now been successfully resolved.
